README.md

```diff
@@ -97,45 +95,91 @@ _Only when accessing a property of this module, would the execution operations b
 This way, `DeferredModuleNamespace` acts like a proxy to the evaluation of the module, effectively
```

Comment:
Previously I imagined `DeferredModuleNamespace` as a special variation of namespaces, I believe that is no longer the intent? Perhaps we can update this.